src/main/webapp/3Dscript.js

```diff
@@ -351,6 +352,25 @@ function makeGUI() {
   roll = gui.add(rollStart, 'roll');
 }
 
+/** Updates time gui to show if time typed in is found or not. */
+function findTime() {
+  let found = false;
+  console.log(time.getValue());
+  for (let i = 0; i< poseLength; i++) {
+    // If point with time stamp is found
+    if (pose[i].gpsTimestamp == time.getValue()) {
+      unselectCylinder();
+      selectedIndex = i;
+      selectCylinder();
+      displayPointValues(i);
+      found= true;
+    }
+  }
+  if (!found) {
+    time.setValue('no time found');
```

Comment:
You can return the data which timestamp >= input timestamp instead of the exactly the same time to make it more friendly to users, usually users will search an ambiguous timestamp and want to find the nearby data.